### PR TITLE
add role files

### PR DESCRIPTION
**Pull request by ghost**
@clwells ^^

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+- name: Install Debian-cd Update CRON
+  template:
+    src="debian-cd-mirror.j2"
+    dest="{{ mirror_cron_dir }}/debian-cd-mirror"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Install cd-mirror script
```
Comment (clwells) [inline tasks/main.yml]: capital "Script"

File: tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  copy:
+    src="cd-mirror"
+    dest="{{ mirror_bin_dir }}/cd-mirror"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Install cd-mirror config file
```
Comment (clwells) [inline tasks/main.yml]: Capital 'Config File'

File: tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  template:
+    src="debian-cd-mirror.conf.j2"
+    dest="{{ mirror_etc_dir }}/debian-cd-mirror.conf"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Install ftpsync script
```
Comment (clwells) [inline tasks/main.yml]: capital "Script"

File: tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+- name: Install ftpsync config file
+  template:
+    src="ftpsync.conf.j2"
+    dest="{{ mirror_etc_dir }}/ftpsync.conf"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Install jigdo-mirror script
```
Comment (clwells) [inline tasks/main.yml]: capital "Script"

File: tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+- name: Install ftpsync script
+  copy:
+    src="ftpsync"
+    dest="{{ mirror_bin_dir }}/ftpsync"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Install ftpsync config file
```
Comment (clwells) [inline tasks/main.yml]: Capital 'Config File'

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+- name: Install jigdo-mirror script
+  copy:
+    src="jigdo-mirror"
+    dest="{{ mirror_bin_dir }}/jigdo-mirror"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Install runmirrors script
```
Comment (clwells) [inline tasks/main.yml]: capital "Script"

File: tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+  copy:
+    src="runmirrors"
+    dest="{{ mirror_bin_dir }}/runmirrors"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Install common script
```
Comment (clwells) [inline tasks/main.yml]: capital "Script"

File: tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  copy:
+    src="common"
+    dest="{{ mirror_etc_dir }}/common"
+    owner="{{ mirror_unix_user }}"
+    group="{{ mirror_unix_group }}"
+    mode=0700
+
+- name: Ensure jigdo config directory exists
```
Comment (clwells) [inline tasks/main.yml]: Capital 'Jigdo Config Directory Exists'

File: tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+  file:
+    dest="{{ mirror_etc_dir }}/jigdo"
+    state="directory"
+
+- name: Copy each jigdo config over
```
Comment (clwells) [inline tasks/main.yml:80]: Capital 'Each Jigdo Config Over'

**Comment by clwells** [comment]
How much space do all of the archs take up?  Should we restrict to certain arches?

**Comment by ghost** [comment]
@clwells it takes 1.5TB to mirror the entire archive, our servers has 16TB total. I personally wouldn't mirror it at all since we don't use it internally not due to the amount of space it takes up; Since support is reason why we do mirror it, we could ping them about what archs they actually care about and delete the rest if you think 1.5TB is too much.

**Comment by ghost** [comment]
to add context, mirroring fedora takes 1.7TB with less archs

**Comment by clwells** [comment]
Y, please ping them.  If it's all that's fine. If it's less that's fine too.

**Comment by ghost** [comment]
The ftpsync script doesn't have an way to selectively sync only certain archs, so we are keeping everything.

File: tasks/main.yml (hunk 9 above)
Comment (clwells) [reply tasks/main.yml:80]: @nexecook ^^

**Pull request merged**